DirectProgramming/DPC++FPGA/ReferenceDesigns/db/src/query11/query11_kernel.cpp

```diff
@@ -120,33 +119,36 @@ bool SubmitQuery11(queue& q, Database& dbinfo, std::string& nation,
   auto join_event = q.submit([&](handler& h) {
     // SUPPLIER table accessors
     size_t s_rows = dbinfo.s.rows;
-    accessor s_suppkey_accessor(s_suppkey_buf, h, read_only);
     accessor s_nationkey_accessor(s_nationkey_buf, h, read_only);
 
     h.single_task<JoinPartSupplierParts>([=]() [[intel::kernel_args_restrict]] {
       // initialize the array map
       // +1 is to account for fact that SUPPKEY is [1,kSF*10000]
-      using ArrayMapType = ArrayMap<SupplierRow, kSupplierTableSize + 1>;
-      ArrayMapType array_map;
-      array_map.Init();
+      unsigned char nation_key_map_data[kSupplierTableSize + 1];
+      bool nation_key_map_valid[kSupplierTableSize + 1];
+      for (int i = 0; i < kSupplierTableSize + 1; i++) {
+        nation_key_map_valid[i] = false;
+      }
 
       // populate MapJoiner map
       // why a map? keys may not be sequential
-      [[intel::initiation_interval(1)]]
+      [[intel::initiation_interval(1), intel::ivdep]]
```

Comment:
I think a comment here explaining why ivdep is safe is justified.

Also, can we apply the ivdep to just a single variable?  I thought blanket ivdep was discouraged.

Reply:
> I think a comment here explaining why ivdep is safe is justified.
Agreed. I will do so.

>Also, can we apply the ivdep to just a single variable?
I need to apply it to two variables. Maybe I can try `[[intel::initiation_interval(1), intel::ivdep(var1), intel::ivdep(var2)]]` ?

Reply:
Actually. I just realized I don't even need ivdep since the supplier keys are sequential, I can use the loop index variable as the index into the map and the compiler can determine that there is no loop carried deps itself. I will remove them.